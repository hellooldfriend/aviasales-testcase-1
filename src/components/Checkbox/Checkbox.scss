@import '../../variables.scss';

.checkbox {
    display: inline-block;
    width: 100%;
    box-sizing: border-box;
    padding: $p1 $p2 $p1 50px;
    position: relative;

    &:hover {
        background: $background_hover;
        cursor: pointer;
    }

    input {
        position: absolute;
        left: 0;
        opacity: 0;
    }




    &:before {
        content: '';
        display: inline-block;
        width: 20px;
        height: 20px;
        border: 1px solid $border_color_basic;
        border-radius: 4px;
        position: absolute;
        left: 20px;
        top: 0;
        bottom: 0;
        margin-top: auto;
        margin-bottom: auto;
    }

    &-check {
        display: inline-block;
        position: absolute;
        left: 25px;
        opacity: 0;
    }


    &._disabled {
        cursor: default;

        &:before {
            border-color: $border_color_disabled;
        }
    }

    &._checked {
        .checkbox-check {
            opacity: 1;
        }
    }
}