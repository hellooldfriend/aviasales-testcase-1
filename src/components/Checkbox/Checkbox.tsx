import React from 'react'
import './Checkbox.scss'

import { ICheckbox } from '../../types'
import { IconCheck } from '../../icons'


const Checkbox = (props: ICheckbox) => {
    const cls = ['checkbox']

    if(props.disabled) {
        cls.push('_disabled')
    }
    if(props.value) {
        cls.push('_checked')
    }

    return (
        <label className={cls.join(' ')}>

            {props.value &&
                <span className="checkbox-check">
                    <IconCheck />
                </span>
            }

            <input
                type="checkbox"
                checked={props.value}
                onChange={() => props.onChange(!props.value)}
                disabled={props.disabled}
            />
            {props.title}
        </label>
    )
}


export default Checkbox