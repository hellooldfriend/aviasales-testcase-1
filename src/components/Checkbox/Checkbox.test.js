import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import Checkbox from './Checkbox'


afterEach(() => {
    cleanup()
})

const props = {
    value: true,
    title: 'Test',
}


test('Should render active Checkbox', () => {
    render(<Checkbox value={props.value} title={props.title} />)
    const element = screen.getByTestId('checkbox')
    expect(element).toBeInTheDocument()
    expect(element).toHaveClass('_checked')
    expect(element).toHaveTextContent(props.title)
})

test('Should render not active Checkbox', () => {
    render(<Checkbox value={false} title={props.title} />)
    const element = screen.getByTestId('checkbox')
    expect(element).toBeInTheDocument()
    expect(element).not.toHaveClass('_checked')
    expect(element).toHaveTextContent(props.title)
})

test('Should render disabled Checkbox', () => {
    render(<Checkbox disabled={true} title={props.title} />)
    const element = screen.getByTestId('checkbox')
    expect(element).toBeInTheDocument()
    expect(element).toHaveClass('_disabled')
    expect(element).toHaveTextContent(props.title)
})


test('Matches snapshot', () => {
    const tree = renderer.create(<Checkbox value={true} title={props.title} />).toJSON()
    expect(tree).toMatchSnapshot()
})