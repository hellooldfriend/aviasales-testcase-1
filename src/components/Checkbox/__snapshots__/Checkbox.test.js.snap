// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<label
  className="checkbox _checked"
  data-testid="checkbox"
>
  <span
    className="checkbox-check"
  >
    <svg
      fill="none"
      height="8"
      viewBox="0 0 12 8"
      width="12"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4.28571 8L0 4.16123L1.20857 3.0787L4.28571 5.82726L10.7914 0L12 1.09021L4.28571 8Z"
        fill="#2196F3"
      />
    </svg>
  </span>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  Test
</label>
`;
